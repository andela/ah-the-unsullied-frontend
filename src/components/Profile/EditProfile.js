import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Row, Modal, Input } from 'react-materialize';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import * as profileActions from '../../actions/profile/profile';
import ImageUploaoder from '../../utils/ImageUploaoder';
import '../../assets/styles/HomePage.scss';

class ViewProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bio: "",
      image: props.image
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit = e => {
    e.preventDefault();
    const { bio, image } = this.state;
    const data = {
      bio,
      image
    };
    const { actions } = this.props;
    actions.EditUserProfile('sammy', data);
  };

  fileHandler = e => {
    const selectFile = e.target.files[0];
    ImageUploaoder({ image: selectFile }).then(res =>
      this.setState({
        image: res.data.secure_url
      })
    );
  };

  onChange(e) {
    this.setState({
      bio: e.target.value
    });
  }

  render() {
    return (
      <div>
        <Modal
          header="Profile Edit"
          className="editPage"
          fixedFooter
          trigger={
            <button className="btn waves-effect edit">
              Edit profile
              <i className="material-icons right">edit</i>
            </button>
          }
        >
          <Row>
            <img
              src={this.props.image}
              className="responsive-img circle editPageProfile"
              onClick={() => this.fileInput.click()}
              height="100px"
              width="100px"
              alt="Avatar"
            />
          </Row>
          <Row>
            <form onSubmit={this.handleSubmit}>
              <label htmlFor="bio">Bio</label>
              <textarea
                id="bio"
                className="materialize-textarea"
                onChange={this.onChange}
                defaultValue={this.props.bio}
              />
              <Row>
                <Input
                  s={12}
                  type="file"
                  name="image"
                  ref={fileInput => {
                    this.fileInput = fileInput;
                  }}
                  label="Upload Image"
                  onChange={this.fileHandler}
                  defaults={12}
                />
              </Row>
              <Row className="left-align">
                <button className="btn waves-effect edit " type="submit">
                  Update
                </button>
              </Row>
            </form>
          </Row>
        </Modal>
      </div>
    );
  }
}

ViewProfile.propTypes = {
  actions: PropTypes.func.isRequired,
  bio: PropTypes.string.isRequired
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(profileActions, dispatch)
  };
}

function mapStateToProps(state) {
  return {
    UserProfile: state.profile
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewProfile);
