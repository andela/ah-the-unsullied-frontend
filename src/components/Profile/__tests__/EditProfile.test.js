import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import EditProfile from '../EditProfile.js';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    registerUser: jest.fn()
  };
  const wrapper = shallow(<EditProfile {...props} />);

  return {
    props,
    wrapper
  };
}

describe('Edit Profile tests', () => {
  it('renders the edit form', () => {
    const { wrapper } = setup();
  });
});
