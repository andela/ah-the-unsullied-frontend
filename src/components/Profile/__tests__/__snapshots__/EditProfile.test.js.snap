// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get profile should render article view component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewProfile
    EditUserProfile={[MockFunction]}
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    bio="a bio "
    image="hshshs"
    match={
      Object {
        "params": Object {
          "username": "nesh",
        },
      }
    }
    username="nesh"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <Modal
          actions={
            Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
              >
                Close
              </Button>,
            ]
          }
          bottomSheet={false}
          className="editPage"
          fixedFooter={true}
          header="Profile Edit"
          modalOptions={Object {}}
          trigger={
            <button
              className="btn waves-effect edit"
            >
              Edit profile
              <i
                className="material-icons right"
              >
                edit
              </i>
            </button>
          }
        >
          <Row>
            <img
              alt="Avatar"
              className="responsive-img circle editPageProfile"
              height="100px"
              src="hshshs"
              width="100px"
            />
          </Row>
          <Row>
            <form
              onSubmit={[Function]}
            >
              <label
                htmlFor="bio"
              >
                Bio
              </label>
              <textarea
                className="materialize-textarea"
                defaultValue="a bio "
                id="bio"
                onChange={[Function]}
              />
              <Row>
                <Input
                  checked={false}
                  defaults={12}
                  label="Upload Image"
                  name="image"
                  onChange={[Function]}
                  s={12}
                  type="file"
                />
              </Row>
              <Row
                className="left-align"
              >
                <button
                  className="btn waves-effect edit "
                  type="submit"
                >
                  Update
                </button>
              </Row>
            </form>
          </Row>
        </Modal>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          actions={
            Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
              >
                Close
              </Button>,
            ]
          }
          bottomSheet={false}
          className="editPage"
          fixedFooter={true}
          header="Profile Edit"
          modalOptions={Object {}}
          trigger={
            <button
              className="btn waves-effect edit"
            >
              Edit profile
              <i
                className="material-icons right"
              >
                edit
              </i>
            </button>
          }
        >
          <Row>
            <img
              alt="Avatar"
              className="responsive-img circle editPageProfile"
              height="100px"
              src="hshshs"
              width="100px"
            />
          </Row>
          <Row>
            <form
              onSubmit={[Function]}
            >
              <label
                htmlFor="bio"
              >
                Bio
              </label>
              <textarea
                className="materialize-textarea"
                defaultValue="a bio "
                id="bio"
                onChange={[Function]}
              />
              <Row>
                <Input
                  checked={false}
                  defaults={12}
                  label="Upload Image"
                  name="image"
                  onChange={[Function]}
                  s={12}
                  type="file"
                />
              </Row>
              <Row
                className="left-align"
              >
                <button
                  className="btn waves-effect edit "
                  type="submit"
                >
                  Update
                </button>
              </Row>
            </form>
          </Row>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            <Button
              flat={true}
              modal="close"
              node="button"
              waves="light"
            >
              Close
            </Button>,
          ],
          "bottomSheet": false,
          "children": Array [
            <Row>
              <img
                alt="Avatar"
                className="responsive-img circle editPageProfile"
                height="100px"
                src="hshshs"
                width="100px"
              />
            </Row>,
            <Row>
              <form
                onSubmit={[Function]}
              >
                <label
                  htmlFor="bio"
                >
                  Bio
                </label>
                <textarea
                  className="materialize-textarea"
                  defaultValue="a bio "
                  id="bio"
                  onChange={[Function]}
                />
                <Row>
                  <Input
                    checked={false}
                    defaults={12}
                    label="Upload Image"
                    name="image"
                    onChange={[Function]}
                    s={12}
                    type="file"
                  />
                </Row>
                <Row
                  className="left-align"
                >
                  <button
                    className="btn waves-effect edit "
                    type="submit"
                  >
                    Update
                  </button>
                </Row>
              </form>
            </Row>,
          ],
          "className": "editPage",
          "fixedFooter": true,
          "header": "Profile Edit",
          "modalOptions": Object {},
          "trigger": <button
            className="btn waves-effect edit"
          >
            Edit profile
            <i
              className="material-icons right"
            >
              edit
            </i>
          </button>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <img
                alt="Avatar"
                className="responsive-img circle editPageProfile"
                height="100px"
                src="hshshs"
                width="100px"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Avatar",
                "className": "responsive-img circle editPageProfile",
                "height": "100px",
                "src": "hshshs",
                "width": "100px",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <form
                onSubmit={[Function]}
              >
                <label
                  htmlFor="bio"
                >
                  Bio
                </label>
                <textarea
                  className="materialize-textarea"
                  defaultValue="a bio "
                  id="bio"
                  onChange={[Function]}
                />
                <Row>
                  <Input
                    checked={false}
                    defaults={12}
                    label="Upload Image"
                    name="image"
                    onChange={[Function]}
                    s={12}
                    type="file"
                  />
                </Row>
                <Row
                  className="left-align"
                >
                  <button
                    className="btn waves-effect edit "
                    type="submit"
                  >
                    Update
                  </button>
                </Row>
              </form>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="bio"
                  >
                    Bio
                  </label>,
                  <textarea
                    className="materialize-textarea"
                    defaultValue="a bio "
                    id="bio"
                    onChange={[Function]}
                  />,
                  <Row>
                    <Input
                      checked={false}
                      defaults={12}
                      label="Upload Image"
                      name="image"
                      onChange={[Function]}
                      s={12}
                      type="file"
                    />
                  </Row>,
                  <Row
                    className="left-align"
                  >
                    <button
                      className="btn waves-effect edit "
                      type="submit"
                    >
                      Update
                    </button>
                  </Row>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Bio",
                    "htmlFor": "bio",
                  },
                  "ref": null,
                  "rendered": "Bio",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "materialize-textarea",
                    "defaultValue": "a bio ",
                    "id": "bio",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Input
                      checked={false}
                      defaults={12}
                      label="Upload Image"
                      name="image"
                      onChange={[Function]}
                      s={12}
                      type="file"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "checked": false,
                      "defaults": 12,
                      "label": "Upload Image",
                      "name": "image",
                      "onChange": [Function],
                      "s": 12,
                      "type": "file",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <button
                      className="btn waves-effect edit "
                      type="submit"
                    >
                      Update
                    </button>,
                    "className": "left-align",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Update",
                      "className": "btn waves-effect edit ",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Update",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "form",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <Modal
            actions={
              Array [
                <Button
                  flat={true}
                  modal="close"
                  node="button"
                  waves="light"
                >
                  Close
                </Button>,
              ]
            }
            bottomSheet={false}
            className="editPage"
            fixedFooter={true}
            header="Profile Edit"
            modalOptions={Object {}}
            trigger={
              <button
                className="btn waves-effect edit"
              >
                Edit profile
                <i
                  className="material-icons right"
                >
                  edit
                </i>
              </button>
            }
          >
            <Row>
              <img
                alt="Avatar"
                className="responsive-img circle editPageProfile"
                height="100px"
                src="hshshs"
                width="100px"
              />
            </Row>
            <Row>
              <form
                onSubmit={[Function]}
              >
                <label
                  htmlFor="bio"
                >
                  Bio
                </label>
                <textarea
                  className="materialize-textarea"
                  defaultValue="a bio "
                  id="bio"
                  onChange={[Function]}
                />
                <Row>
                  <Input
                    checked={false}
                    defaults={12}
                    label="Upload Image"
                    name="image"
                    onChange={[Function]}
                    s={12}
                    type="file"
                  />
                </Row>
                <Row
                  className="left-align"
                >
                  <button
                    className="btn waves-effect edit "
                    type="submit"
                  >
                    Update
                  </button>
                </Row>
              </form>
            </Row>
          </Modal>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Modal
            actions={
              Array [
                <Button
                  flat={true}
                  modal="close"
                  node="button"
                  waves="light"
                >
                  Close
                </Button>,
              ]
            }
            bottomSheet={false}
            className="editPage"
            fixedFooter={true}
            header="Profile Edit"
            modalOptions={Object {}}
            trigger={
              <button
                className="btn waves-effect edit"
              >
                Edit profile
                <i
                  className="material-icons right"
                >
                  edit
                </i>
              </button>
            }
          >
            <Row>
              <img
                alt="Avatar"
                className="responsive-img circle editPageProfile"
                height="100px"
                src="hshshs"
                width="100px"
              />
            </Row>
            <Row>
              <form
                onSubmit={[Function]}
              >
                <label
                  htmlFor="bio"
                >
                  Bio
                </label>
                <textarea
                  className="materialize-textarea"
                  defaultValue="a bio "
                  id="bio"
                  onChange={[Function]}
                />
                <Row>
                  <Input
                    checked={false}
                    defaults={12}
                    label="Upload Image"
                    name="image"
                    onChange={[Function]}
                    s={12}
                    type="file"
                  />
                </Row>
                <Row
                  className="left-align"
                >
                  <button
                    className="btn waves-effect edit "
                    type="submit"
                  >
                    Update
                  </button>
                </Row>
              </form>
            </Row>
          </Modal>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actions": Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
              >
                Close
              </Button>,
            ],
            "bottomSheet": false,
            "children": Array [
              <Row>
                <img
                  alt="Avatar"
                  className="responsive-img circle editPageProfile"
                  height="100px"
                  src="hshshs"
                  width="100px"
                />
              </Row>,
              <Row>
                <form
                  onSubmit={[Function]}
                >
                  <label
                    htmlFor="bio"
                  >
                    Bio
                  </label>
                  <textarea
                    className="materialize-textarea"
                    defaultValue="a bio "
                    id="bio"
                    onChange={[Function]}
                  />
                  <Row>
                    <Input
                      checked={false}
                      defaults={12}
                      label="Upload Image"
                      name="image"
                      onChange={[Function]}
                      s={12}
                      type="file"
                    />
                  </Row>
                  <Row
                    className="left-align"
                  >
                    <button
                      className="btn waves-effect edit "
                      type="submit"
                    >
                      Update
                    </button>
                  </Row>
                </form>
              </Row>,
            ],
            "className": "editPage",
            "fixedFooter": true,
            "header": "Profile Edit",
            "modalOptions": Object {},
            "trigger": <button
              className="btn waves-effect edit"
            >
              Edit profile
              <i
                className="material-icons right"
              >
                edit
              </i>
            </button>,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="Avatar"
                  className="responsive-img circle editPageProfile"
                  height="100px"
                  src="hshshs"
                  width="100px"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Avatar",
                  "className": "responsive-img circle editPageProfile",
                  "height": "100px",
                  "src": "hshshs",
                  "width": "100px",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <form
                  onSubmit={[Function]}
                >
                  <label
                    htmlFor="bio"
                  >
                    Bio
                  </label>
                  <textarea
                    className="materialize-textarea"
                    defaultValue="a bio "
                    id="bio"
                    onChange={[Function]}
                  />
                  <Row>
                    <Input
                      checked={false}
                      defaults={12}
                      label="Upload Image"
                      name="image"
                      onChange={[Function]}
                      s={12}
                      type="file"
                    />
                  </Row>
                  <Row
                    className="left-align"
                  >
                    <button
                      className="btn waves-effect edit "
                      type="submit"
                    >
                      Update
                    </button>
                  </Row>
                </form>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="bio"
                    >
                      Bio
                    </label>,
                    <textarea
                      className="materialize-textarea"
                      defaultValue="a bio "
                      id="bio"
                      onChange={[Function]}
                    />,
                    <Row>
                      <Input
                        checked={false}
                        defaults={12}
                        label="Upload Image"
                        name="image"
                        onChange={[Function]}
                        s={12}
                        type="file"
                      />
                    </Row>,
                    <Row
                      className="left-align"
                    >
                      <button
                        className="btn waves-effect edit "
                        type="submit"
                      >
                        Update
                      </button>
                    </Row>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Bio",
                      "htmlFor": "bio",
                    },
                    "ref": null,
                    "rendered": "Bio",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "materialize-textarea",
                      "defaultValue": "a bio ",
                      "id": "bio",
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <Input
                        checked={false}
                        defaults={12}
                        label="Upload Image"
                        name="image"
                        onChange={[Function]}
                        s={12}
                        type="file"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "checked": false,
                        "defaults": 12,
                        "label": "Upload Image",
                        "name": "image",
                        "onChange": [Function],
                        "s": 12,
                        "type": "file",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <button
                        className="btn waves-effect edit "
                        type="submit"
                      >
                        Update
                      </button>,
                      "className": "left-align",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Update",
                        "className": "btn waves-effect edit ",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Update",
                      "type": "button",
                    },
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
